File: src/fs/IUBH.TOR/IUBH.TOR.iOS/DefaultViewCellRenderer.fs
namespace IUBH.TOR.iOS

open UIKit
open Xamarin.Forms
open Xamarin.Forms.Platform.iOS

/// The default behavior of XF will leave us with a selected
/// cell that can't be unselected. This custom renderer fixes that.
type DefaultViewCellRenderer() =
    inherit ViewCellRenderer()

    override this.GetCell(item, reusableCell, tableView) =
      let cell = base.GetCell(item, reusableCell, tableView)
      cell.SelectionStyle <- UITableViewCellSelectionStyle.None
      cell

[<assembly: ExportRendererAttribute (typeof<ViewCell>, typeof<DefaultViewCellRenderer>)>] do ()

File: src/cs/IUBH.TOR/IUBH.TOR.iOS/CustomRenderers/DefaultViewCellRenderer.cs
using IUBH.TOR.iOS.CustomRenderers;
using UIKit;
using Xamarin.Forms;
using Xamarin.Forms.Platform.iOS;

[assembly: ExportRenderer(typeof(ViewCell), typeof(DefaultViewCellRenderer))]

namespace IUBH.TOR.iOS.CustomRenderers
{
    /// <summary>
    /// Makes sure cells in lists do not appear to be selected. This solves
    /// an ugly glitch in the current Xamarin.Forms implementation.
    /// </summary>
    public class DefaultViewCellRenderer : ViewCellRenderer
    {
        public override UITableViewCell GetCell(
            Cell item,
            UITableViewCell reusableCell,
            UITableView tv
        )
        {
            var cell = base.GetCell(item, reusableCell, tv);

            cell.SelectionStyle = UITableViewCellSelectionStyle.None;

            return cell;
        }
    }
}
